### PR TITLE
Implement integer conversion functions and add corresponding tests

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Area of change
- Implement integer conversion functions and add corresponding tests
```

**Int8:** fromUInt8, fromInt16, fromUInt16, fromInt32, fromUInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**UInt8:** fromInt8, fromInt16, fromUInt16, fromInt32, fromUInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**Int16:** fromInt8, fromUInt8, fromUInt16, fromInt32, fromUInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**UInt16:** fromInt8, fromUInt8, fromInt16, fromInt32, fromUInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**Int32:** fromInt8, fromUInt8, fromInt16, fromInt16, fromUInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**UInt32:** fromInt8, fromUInt8, fromInt16, fromInt16, fromInt32, fromInt64, fromUInt64, fromInt128, fromUInt128
**Int64:** fromInt8, fromUInt8, fromInt16, fromInt16, fromInt32, fromUInt32, fromUInt64, fromInt128, fromUInt128
**UInt64:** fromInt8, fromUInt8, fromInt16, fromInt16, fromInt32, fromUInt32, fromInt64, fromInt128, fromUInt128
**Int128:** fromInt8, fromUInt8, fromInt16, fromInt16, fromInt32, fromUInt32, fromInt64, fromUInt64
**UInt128:** fromUInt8, fromUInt16, fromUInt32, fromUInt64

### CODE REVIEW

File: backend/src/Prelude/Prelude.fs [hunk 1]
```diff
@@ -397,3 +397,18 @@ let randomString (length : int) : string =
 
   assertEq "randomString length is correct" result.Length length
   result
+
+
+module NumericLiteralQ =
```
Comment (OceanOak) [inline backend/src/Prelude/Prelude.fs:402]: I am not sure if this is the right place for these two modules, we need them in all Builtin (U)IntX files and in TestUtils. Please let me know if I should move them elsewhere

**Review by pbiggar — CHANGES_REQUESTED** [review]
I think there are some errors in the conversions. In particular, when going from type A which fully fits in type B (eg int16 to int32), we shouldn't need an option.

**Review by pbiggar — APPROVED** [review]
OK, this looks good!

File: backend/testfiles/execution/stdlib/int8.dark [hunk 2]
```diff
@@ -342,6 +342,80 @@ PACKAGE.Darklang.Stdlib.Int8.parse_v0 "XIV" = PACKAGE.Darklang.Stdlib.Result.Res
...
+  127y
+
+PACKAGE.Darklang.Stdlib.Int8.fromInt32_v0 128l = PACKAGE.Darklang.Stdlib.Option.Option.None
+
+PACKAGE.Darklang.Stdlib.Int8.fromInt32_v0 (-1l) = PACKAGE.Darklang.Stdlib.Option.Option.Some
+  -1y
+
+PACKAGE.Darklang.Stdlib.Int8.fromInt32_v0 (-128l) = PACKAGE.Darklang.Stdlib.Option.Option.Some
+  -128y
+
```
Comment (OceanOak) [inline backend/testfiles/execution/stdlib/int8.dark:404]: Followup PR 
PACKAGE.Darklang.Stdlib.Int8.fromInt32_v0 (-129l) = PACKAGE.Darklang.Stdlib.Option.Option.None

File: backend/testfiles/execution/stdlib/uint64.dark [hunk 3]
```diff
@@ -213,4 +213,81 @@ PACKAGE.Darklang.Stdlib.UInt64.parse_v0 "XIV" = PACKAGE.Darklang.Stdlib.Result.R
...
+
+
+PACKAGE.Darklang.Stdlib.UInt64.fromInt8_v0 0y = PACKAGE.Darklang.Stdlib.Option.Option.Some
+  0UL
+
+PACKAGE.Darklang.Stdlib.UInt64.fromInt8_v0 1y = PACKAGE.Darklang.Stdlib.Option.Option.Some
+  1UL
+
+PACKAGE.Darklang.Stdlib.UInt64.fromInt8_v0 127y = PACKAGE.Darklang.Stdlib.Option.Option.Some
+  127UL
```
Comment (OceanOak) [inline backend/testfiles/execution/stdlib/uint64.dark:226]: Followup PR add 
PACKAGE.Darklang.Stdlib.UInt64.fromInt8_v0 -128y = PACKAGE.Darklang.Stdlib.Option.Option.None

**Pull request merged**